src/Razor/src/Microsoft.AspNetCore.Razor.Common/Telemetry/TelemetryReporter.cs

```diff
@@ -73,4 +118,47 @@ private void Report(TelemetryEvent telemetryEvent)
             _logger?.LogError(e, "Failed logging telemetry event");
         }
     }
+
+    private static string GetDescription(Exception exception)
+    {
+        const string CodeAnalysisNamespace = nameof(Microsoft) + "." + nameof(CodeAnalysis);
+        const string AspNetCoreNamespace = nameof(Microsoft) + "." + nameof(AspNetCoreNamespace);
+
+        // Be resilient to failing here.  If we can't get a suitable name, just fall back to the standard name we
+        // used to report.
+        try
+        {
+            // walk up the stack looking for the first call from a type that isn't in the ErrorReporting namespace.
+            var frames = new StackTrace(exception).GetFrames();
+
+            // On the .NET Framework, GetFrames() can return null even though it's not documented as such.
+            // At least one case here is if the exception's stack trace itself is null.
+            if (frames != null)
+            {
+                foreach (var frame in frames)
+                {
+                    var method = frame?.GetMethod();
+                    var methodName = method?.Name;
+                    if (methodName == null)
+                        continue;
```

Comment:
Nit: I don't think we use optional brackets anywhere else in this repo so lets keep that consistent and add some brackets for these.